encoders/numeric/CompressionVaeEncoder/__init__.py

```diff
@@ -16,68 +17,60 @@ class CompressionVaeEncoder(BaseNumericEncoder, BaseTFEncoder):
-        :param initializer: controls how the model weights are initialized, with options being `orthogonal` (default),
-        `truncated_normal`, and `lecun_normal`
-        :param batch_size: determine the batch sizes used for training.
-        :param batch_size_test: determine the batch sizes used for testing.
-        :param feature_normalization: tells CVAE whether it should internally apply feature normalization
-        (zero mean, unit variance, based on the training data) or not. If True, the normalisation factors are stored
-        with the model and get applied to any future data.
-        :param tb_logging: determines whether the model writes summaries for TensorBoard during the training process.
+        :param model_path: specifies the path to the pretrained model
+
         """
         super().__init__(*args, **kwargs)
-        self.X = X
-        self.train_valid_split = train_valid_split
-        self.output_dim = output_dim
-        self.iaf_flow_length = iaf_flow_length
-        self.cells_encoder = cells_encoder
-        self.initializer = initializer
-        self.batch_size = batch_size
-        self.batch_size_test = batch_size_test
         self.model_path = model_path
-        self.feature_normalization = feature_normalization
-        self.tb_logging = tb_logging
 
     def post_init(self):
-        super().post_init()
-        self.to_device()
+        from cvae import cvae
+        import cvae.lib.model_iaf as model
+        import tensorflow as tf
+
+        config = tf.ConfigProto(log_device_placement=False)
+        config.gpu_options.allow_growth = True
+
+        with tf.Graph().as_default():
+            # Load parameter file.
+            with open(f'{self.model_path}/params.json', 'r') as f:
+                param = json.load(f)
+
+            net = model.VAEModel(param,
+                                 None,
+                                 input_dim=param['dim_feature'],
+                                 keep_prob=tf.placeholder_with_default(input=tf.cast(1.0, dtype=tf.float32),
+                                                                       shape=(),
+                                                                       name="KeepProb"),
+                                 initializer='orthogonal')
+            # Placeholder for data features
+            self.data_feature_placeholder = tf.placeholder_with_default(
+                input=tf.zeros([64, param['dim_feature']], dtype=tf.float32),
+                shape=[None, param['dim_feature']])
+
+            self.embeddings = net.embed(self.data_feature_placeholder)
 
-    @batching
+            self.sess = tf.Session(config=config)
+            init = tf.global_variables_initializer()
+            self.sess.run(init)
+
+            # Saver for loading checkpoints of the model.
+            saver = tf.train.Saver(var_list=tf.trainable_variables(), max_to_keep=2)
```

Comment:
Wjat isthis max to keep param?

Reply:
It's the maximum number of model checkpoints to keep. It's used in training but here is useless, I'm going to drop it.